### PR TITLE
Verify SVCUID before providing access tokens via TES

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: ggipcd/unit/ggl.gg-ipc.socket.socket.in [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[Unit]
```
Comment (aws-kevinrickard) [inline ggipcd/unit/ggl.gg-ipc.socket.socket.in:1]: `socket.socket` in the file name, is that intentional?

File: tes-serverd/src/http_server.c [hunk 2]
```diff
@@ -72,7 +74,65 @@ static void request_handler(struct evhttp_request *req, void *arg) {
         }
         return;
     }
-    // TODO: Check the Authorization's value
+
+    GGL_LOGI("Auth header received: %s", auth_header);
```
Comment (aws-kevinrickard) [inline tes-serverd/src/http_server.c:78]: Is writing the auth header to logs okay? Seems like then anyone with access to read the logs can then spoof a request

Comment (AniruddhaKanhere) [reply tes-serverd/src/http_server.c:77]: I can remove that . Good catch

File: ggipcd/unit/ggl.gg-ipc.socket.socket.in (hunk 1 above)
Comment (AniruddhaKanhere) [reply ggipcd/unit/ggl.gg-ipc.socket.socket.in:1]: I don't think it is required. It is a typo. I will fix it.

File: tes-serverd/src/http_server.c (hunk 2 above)
Comment (AniruddhaKanhere) [reply tes-serverd/src/http_server.c]: updated

File: ggipcd/src/ipc_components.c [hunk 3]
```diff
@@ -67,11 +75,101 @@ static void set_component_name(
...
+    GGL_LOGE("Requested svcuid not found.");
+
+    ggl_respond(handle, GGL_OBJ_BOOL(false));
+    return GGL_ERR_OK;
+}
+
+static void *ggl_ipc_component_server(void *args) {
+    (void) args;
+
+    // Hack. Fix it. Need IPC to be up and running before we start this server.
```
Comment (aws-kevinrickard) [inline ggipcd/src/ipc_components.c:126]: Adding `TODO` might make this easier to find later

Comment (archigup) [reply ggipcd/src/ipc_components.c:126]: is it even needed? it shouldnt depend on the other thread

Comment (AniruddhaKanhere) [reply ggipcd/src/ipc_components.c]: removed.

File: ggipcd/unit/ggl.gg-ipc.socket.socket.in (hunk 1 above)
Comment (AniruddhaKanhere) [reply ggipcd/unit/ggl.gg-ipc.socket.socket.in:1]: I reverted this change. As the file name is socketName.socket.
Here in this case the socketName is ggipc.socket. That is why the file name is ggipc.socket.socket

**Review by aws-kevinrickard — DISMISSED** [review]

**Pull request merged**